pkg/bot.go

```diff
@@ -113,9 +118,36 @@ type kubeClients struct {
 	// core API client and lister
 	coreClientSet *kubernetes.Clientset
 	podLister     corelister.PodLister
+
+	// gRPC auth admin service
+	authClient pbAdmin.AdministrationClient
+}
+
+func authAdminAPI(k *kubernetes.Clientset, authSVCLabels map[string]string) (pbAdmin.AdministrationClient, error) {
+	svcs, err := k.CoreV1().Services("").List(metav1.ListOptions{LabelSelector: labels.FormatLabels(authSVCLabels)})
```

Comment:
initially, this was DNS but per request from @drekle, changed it to "service discovery". Also if you look a few lines below, it will error out if there are more than one so we can look at our deployment.